R/summarize_copilot.R

```diff
@@ -329,14 +334,14 @@ map_responses_to_cycles <- function(response_tbl,
   # considering the minimum possible set of cycles.
   potential_cycles <- cycle_extended %>%
     filter(cycle.team_id %in% unique(response_merged$classroom.team_id)) %>%
-    filter(!util$is_blank(cycle.start_date))
+    filter(!util$is_blank(cycle.extended_start_date))
```

Comment:
nit: I think this filter is correct, but it could probably use a comment, e.g.

> This considers 3 cases:
> 1. This is an ordinal = 1 cycle, where `extended_start_date` was set by Copilot. This is kept by the filter. Even if this cycle's other dates are unset, we will want assign responses to it by default.
> 2. This is an ordinal > 1 cycle where `start_date` was set by the user. Kept by the filter because code above copied this value into `extended_start_date`.
> 3. This is an ordinal > 1 cycle where neither `start_date` nor `extended_start_date` were are set. These are excluded by the filter because we never want to assign responses to them.

Reply:
99c72dd